pages/[id]/bridge.tsx

```diff
@@ -219,6 +220,81 @@ const Bridge = () => {
   const isAmountError = errorType === ErrorType.ERC20AmountError;
   const isEthError = errorType === ErrorType.InsufficientNativeCurrencyError;
 
+  const connectButton = (
+    <div className="mt-5">
+      <ConnectButton />
+    </div>
+  );
+
+  const switchNetworkButton = (
+    <button
+      type="button"
+      className="mt-5 rounded-md bg-indigo-600 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600 disabled:bg-gray-300 disabled:cursor-not-allowed"
+      onClick={async () => {
+        if (walletClient) switchChain(walletClient, source.chain);
+      }}
+      disabled={walletIsLoading}
+    >
+      Switch network to {source.chain.name}
+    </button>
+  );
+
+  const setAllowanceButton = (
+    <button
+      type="button"
+      className="flex mx-auto mt-5 rounded-md bg-indigo-600 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600 disabled:bg-gray-300 disabled:cursor-not-allowed"
+      onClick={handleAllowance}
+      disabled={!approveL1 || approveL1IsLoading || approvalL1TxIsLoading || !isSufficientBalance}
+    >
+      Set allowance for {source.token?.symbol} on {target.chain.name}
+      {(approveL1IsLoading || approvalL1TxIsLoading) && (
+        <div className="ml-2">
+          <Spinner />
+        </div>
+      )}
+    </button>
+  );
+
+  const bridgeButton = (
+    /* ⚪️ Finally, we can show the bridge button. */
+    <button
+      type="button"
+      className="mt-5 mx-auto flex flex-row items-center rounded-md bg-indigo-600 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600 disabled:bg-gray-300 disabled:cursor-not-allowed"
+      onClick={handleBridge}
+      disabled={
+        rawAmount === BigInt(0) ||
+        (bridgeTarget === BridgeTarget.L2
+          ? !bridgeToL2 || bridgeToL2IsLoading || isEthError
+          : !bridgeToL1 || bridgeToL1IsLoading || isEthError)
+      }
+    >
+      Bridge to {target.chain.name}
+      {(bridgeToL2IsLoading || bridgeToL1IsLoading) && (
+        <span className="ml-2">
+          <Spinner />
+        </span>
+      )}
+    </button>
+  );
+
+  const actionButton = () => {
+    if (mounted) {
```

Comment:
nit: in the case that `mounted === false`, it looks like this method won't return anything. that's probably ok, just calling it out

Reply:
yea, i think it's better not to render anything until we know what's the correct thing to render.